inference-engine/src/mkldnn_plugin/emitters/jit_snippets_emitters.hpp

```diff
@@ -0,0 +1,573 @@
+              const MKLDNNPlugin::emitter_context *emit_context) const override {
+        if (host_isa_ == dnnl::impl::cpu::x64::sse41) {
+            emit_isa<dnnl::impl::cpu::x64::sse41>(in, out);
+        } else if (host_isa_ == dnnl::impl::cpu::x64::avx2) {
+            emit_isa<dnnl::impl::cpu::x64::avx2>(in, out);
+        } else if (host_isa_ == dnnl::impl::cpu::x64::avx512_common) {
+            emit_isa<dnnl::impl::cpu::x64::avx512_common>(in, out);
+        } else {
+            IE_THROW() << host_isa_;
+            assert(!"unsupported isa");
+        }
+    }
+
+    template <dnnl::impl::cpu::x64::cpu_isa_t isa>
+    void emit_isa(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+        using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41,
+                                    Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
+        Reg64 out_reg(ea);
+        Vmm vmm_src0 = Vmm(in[0]);
+        h->uni_vmovups(h->ptr[out_reg], vmm_src0);
+        h->add(out_reg, mkldnn::impl::cpu::x64::cpu_isa_traits<isa>::vlen);
+    }
+};
+
+class ScalarStoreEmitter : public MemoryEmitter {
+public:
+    ScalarStoreEmitter(mkldnn::impl::cpu::x64::jit_generator* h, mkldnn::impl::cpu::x64::cpu_isa_t isa, const std::shared_ptr<ov::Node>& n)
+    : MemoryEmitter(h, isa, n) {
+    }
+
+    size_t get_inputs_num() const override {return 1;}
+
+private:
+    void emit_impl(const std::vector<size_t>& in,
+              const std::vector<size_t>& out,
+              const std::vector<size_t>& pool,
+              const std::vector<size_t>& gpr,
+              const MKLDNNPlugin::emitter_context *emit_context) const override {
+        if (host_isa_ == dnnl::impl::cpu::x64::sse41) {
+            emit_isa<dnnl::impl::cpu::x64::sse41>(in, out);
+        } else if (host_isa_ == dnnl::impl::cpu::x64::avx2) {
+            emit_isa<dnnl::impl::cpu::x64::avx2>(in, out);
+        } else if (host_isa_ == dnnl::impl::cpu::x64::avx512_common) {
+            emit_isa<dnnl::impl::cpu::x64::avx512_common>(in, out);
+        } else {
+            IE_THROW() << host_isa_;
+            assert(!"unsupported isa");
+        }
+    }
+
+    template <dnnl::impl::cpu::x64::cpu_isa_t isa>
+    void emit_isa(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+        using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41,
+                                        Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
+        Reg64 out_reg(ea);
+        Xmm vmm_src0 = Xmm(in[0]);
+        h->uni_vmovss(h->ptr[out_reg], vmm_src0);
+        h->add(out_reg, sizeof(float));
+    }
+};
+
+class LoadEmitter : public MemoryEmitter {
+public:
+    LoadEmitter(mkldnn::impl::cpu::x64::jit_generator* h, mkldnn::impl::cpu::x64::cpu_isa_t isa, const std::shared_ptr<ov::Node>& n)
+    // result of canonizing broadcast by C. Fixe it=)
```

Comment:
TODO

Reply:
Removed this comment and created a common TODO for all memory emitters